crl/checker/checker.go

```diff
@@ -0,0 +1,110 @@
+package checker
+
+import (
+	"bytes"
+	"fmt"
+	"math/big"
+	"sort"
+	"time"
+
+	"github.com/letsencrypt/boulder/crl/crl_x509"
+	"github.com/letsencrypt/boulder/issuance"
+	"github.com/letsencrypt/boulder/linter"
+	crlint "github.com/letsencrypt/boulder/linter/lints/crl"
+)
+
+// Validate runs the given CRL through our set of lints, ensures its signature
+// validates (if supplied with a non-nil issuer), and checks that the CRL is
+// less than ageLimit old. It returns an error if any of these conditions are
+// not met.
+func Validate(crl *crl_x509.RevocationList, issuer *issuance.Certificate, ageLimit time.Duration) error {
+	err := linter.ProcessResultSet(crlint.LintCRL(crl))
+	if err != nil {
+		return fmt.Errorf("linting CRL: %w", err)
+	}
+
+	if issuer != nil {
+		err = crl.CheckSignatureFrom(issuer.Certificate)
+		if err != nil {
+			return fmt.Errorf("checking CRL signature: %w", err)
+		}
+	}
+
+	if time.Since(crl.ThisUpdate) >= ageLimit {
+		return fmt.Errorf("thisUpdate more than %s in the past: %v", ageLimit, crl.ThisUpdate)
+	}
+
+	return nil
+}
+
+type diffResult struct {
+	Added   []*big.Int
```

Comment:
For what it's worth, my crl-monitor code is only using `Removed`.  Unless there's some other implementation that needs Added, I think it's sufficient to just return one.

The code does check for "comparability", but if not for that, we could get Added by passing the same data in opposite orders.

Reply:
Yeah I thought about it a bit, but I think my instinct is to leave it this way, for two reasons:
1) I'd like to keep the compatibility checks, because I've seen too many people make simple mistakes by diffing files in the wrong order, and then drawing the wrong conclusions from the result. I'd like to prevent bugs arising from accidentally diffing CRLs in the wrong order.
2) The original bug filed for this asks for other outputs as well, such as "list of all entries which changed reasonCode", so having a struct to hold that output is forward-compatible.